subprojects/logging/src/test/groovy/org/gradle/internal/featurelifecycle/LoggingDeprecatedFeatureHandlerTest.groovy

```diff
@@ -397,6 +398,27 @@ class LoggingDeprecatedFeatureHandlerTest extends Specification {
         1 * buildOperationListener.progress(_, _) >> { progressFired(it[1], 'feature2') }
     }
 
+    def 'can get removal details for next major Gradle version'() {
+        expect:
+        LoggingDeprecatedFeatureHandler.getRemovalDetails() == "is scheduled to be removed in Gradle ${GradleVersion.current().nextMajor.version}."
+    }
+
+    def 'can get removal details for custom next Gradle version'() {
+        GradleVersion when = GradleVersion.current().getNextMajor().getNextMajor()
+
+        expect:
+        LoggingDeprecatedFeatureHandler.getRemovalDetails(when) == "is scheduled to be removed in Gradle ${when.version}."
+    }
+
+    def 'throws exception when generating removal details for Gradle version lower than current'() {
+        when:
+        LoggingDeprecatedFeatureHandler.getRemovalDetails(GradleVersion.version("2.0"))
+
+        then:
+        def ex = thrown(IllegalArgumentException)
+        ex.message == "Scheduled removal Gradle version has to be >=${GradleVersion.current().version} (given 2.0)"
```

Comment:
Does equal make sense here?  Shouldn't we fail if we emit a deprecation warning that says it'll be removed in the current version?

Reply:
I change the deprecation warning and reverted all these changes.